flink-connectors/flink-connector-hbase-base/src/main/java/org/apache/flink/connector/hbase/sink/HBaseSinkFunction.java

```diff
@@ -201,6 +208,12 @@ public void invoke(T value, Context context) throws Exception {
     }
 
     private void flush() throws IOException {
+        synchronized (mutationMap) {
```

Comment:
Adding mutationMap to drop duplicated data on the client side could not avoid the data consistency issue. For example:
sink.buffer-flush.max-rows=1
+I(1,...)
-U(1,...)
+U(1,...)
These three rows are put into HBase with the same timestamp version.
In the end, HBase cannot find the data with rowkey=1.

Reply:
Hi @YesOrNo828 , this issue has already been addressed in [FLINK-32139](https://github.com/apache/flink/pull/22612) and merged into master.